lib/rouge/lexers/swift.rb

```diff
@@ -73,8 +75,8 @@ class deinit enum convenience extension final func import init internal lazy let
         rule /'(\\.|.)'/, Str::Char
         rule /(\d+\*|\d*\.\d+)(e[+-]?[0-9]+)?/i, Num::Float
         rule /\d+e[+-]?[0-9]+/i, Num::Float
-        rule /0_?[0-7]+(?:_[0-7]+)*/, Num::Oct
-        rule /0x[0-9A-Fa-f]+(?:_[0-9A-Fa-f]+)*/, Num::Hex
+        rule /0o?[0-7]+(?:_[0-7]+)*/, Num::Oct
+        rule /0x[0-9A-Fa-f]+(?:_[0-9A-Fa-f]+)*((\.[0-9A-F]+(?:_[0-9A-F]+)*)?p\d+)?/, Num::Hex
```

Comment:
It looks like from what I can find online (and the visual sample itself) that there should be a `-` after the `p`. Something like this:

```ruby
rule /0x[0-9A-Fa-f]+(?:_[0-9A-Fa-f]+)*((\.[0-9A-F]+(?:_[0-9A-F]+)*)?p-\d+)?/, Num::Hex
```

Reply:
You're right, this is missing an optional sign. This is fixed by 1384b71fac47c3fe0aa8c46b7a0f574afb644a49.

For reference, here's what I could find in the way of authoritative examples of valid and invalid hexadecimal floating-point literals: https://github.com/apple/swift/blob/2ee371b03a0fb0623188d564399681f89e57c36a/test/expr/expressions.swift#L414-L427

Reply:
Awesome! You figure they'd know :)